server/lib/object-storage.ts

```diff
@@ -0,0 +1,153 @@
+import {
+  CompletedPart,
+  CompleteMultipartUploadCommand,
+  CreateMultipartUploadCommand,
+  DeleteObjectCommand,
+  DeleteObjectsCommand,
+  GetObjectCommand,
+  ListObjectsV2Command,
+  PutObjectCommand,
+  S3Client,
+  UploadPartCommand
+} from "@aws-sdk/client-s3"
+import { CONFIG } from "@server/initializers/config"
+import { logger } from '@server/helpers/logger'
+import { createReadStream, createWriteStream, ensureDir, open, close, ReadStream, stat, Stats } from "fs-extra"
+import { Readable } from "stream"
+import { pipeline } from "stream/promises"
+import { dirname } from "path"
+import { min } from "lodash"
+
+type BucketInfo = {BUCKET_NAME: string, PREFIX?: string, BASE_URL?: string}
+const ONE_MIB = 1024 * 1024
+const PART_SIZE = 100 * ONE_MIB
+const MAX_PUT_SIZE = 100 * ONE_MIB
+
+function getS3Client () {
+  return new S3Client({ endpoint: `https://${CONFIG.OBJECT_STORAGE.ENDPOINT}` })
```

Comment:
We'll mock s3 for integration tests (`http://` scheme then) so we should also support URL here (if does not start with `https?://` automatically add `https://` prefix)